File: leetcode/1-two-sum/solution.java
import java.util.HashMap;
import java.util.Map;

class Solution {
    public int[] twoSum(int[] nums, int target) {

        final Map<Integer, Integer> m = new HashMap<>();

        for (int i = 0; i< nums.length; i++) {
            final Integer r = target - nums[i];
            final Integer t = m.get(r);

            if (t != null) {
                return new int[]{t, i};
            }
            
            m.put(nums[i], i);
        }

        return null;
    }
}

File: leetcode/1-two-sum/solution.go
func twoSum(nums []int, target int) []int {
  res := []int{0,0};
  
  m := map[int]int{};
  
  for i := 0; i< len(nums); i++ {
    r := target - nums[i];
    if r, ok := m[r]; ok {
      res[0] = i;
      res[1] = r;
      return res;
    }
    m[nums[i]] = i;

  }
  return res;
   
}
